201-vm-dedicated-hosts/azuredeploy.json

```diff
@@ -0,0 +1,105 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "Location for the resources."
+            }
+        },
+        "numberOfZoness": {
+            "type": "int",
+            "defaultValue": 0,
+            "metadata": {
+                "description": "How many Zone to use. Use 0 for non zonal deployment."
+            }
+        },
+        "numberofHostsPerZone": {
+            "type": "int",
+            "defaultValue": 2,
+            "metadata": {
+                "description": "How many hosts to create per zone."
+            }
+        },
+        "numberOfFDs": {
+            "type": "int",
+            "defaultValue": 2,
+            "metadata": {
+                "description": "How many fault domains to use. "
+            }
+        },
+        "dhgNamePrefix": {
+            "type": "string",
+            "defaultValue": "myHostGroup",
+            "metadata": {
+                "description": "Name (prefix) for your host group."
+            }
+        },
+        "dhNamePrefix": {
+            "type": "string",
+            "defaultValue": "myHost",
+            "metadata": {
+                "description": "Name (prefix) for your host ."
+            }
+        },
+        "dhSKU": {
+            "type": "string",
+            "defaultValue": "DSv3-Type1",
+            "metadata": {
+                "description": "The type (family and generation) for your host ."
+            }
+        }
+    },
+    "variables": {
+        "numberOfHosts": "[if(equals(parameters('numberOfZoness'), 0), parameters('numberofHostsPerZone'), mul(parameters('numberOfZoness'), parameters('numberofHostsPerZone')))]"
+    },
+    "resources": [
+        {
+            "name": "[concat(parameters('dhgNamePrefix'),copyIndex())]",
+            "type": "Microsoft.Compute/HostGroups",
+            "apiVersion": "2018-10-01",
+            "location": "[parameters('location')]",
+            "properties": {
```

Comment:
move properties to the bottom of the resource definition (or move zones and copy above properties)